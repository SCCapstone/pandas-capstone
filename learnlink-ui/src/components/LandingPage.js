import React from 'react';
import './LandingPage.css';
import { FaSearch, FaBell, FaCog, FaUserCircle } from 'react-icons/fa';

const LandingPage = () => {
  return (
    <div className="landing-page">
      <header className="navbar">
        <div className="logo">LearnLink</div>
        <nav className="nav-links">
          <a href="/profile">Profile</a>
          <a href="/messaging">Messaging</a>
          <a href="/resources">Resources</a>
        </nav>
        <div className="nav-icons">
          <div className="search-bar">
            <input type="text" placeholder="search" />
            <FaSearch className="search-icon" />
          </div>
          <FaBell className="icon" />
          <FaCog className="icon" />
          <FaUserCircle className="icon profile-icon" />
        </div>
      </header>
      <main className="content">
        <p>To start searching please update your profile via top left corner!</p>
      </main>
      <footer className="footer">Â© LearnLink</footer>
    </div>
  );
};

export default LandingPage;