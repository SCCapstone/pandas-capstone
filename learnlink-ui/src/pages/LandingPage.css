
  
  .content {
    flex-grow: 1;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 1.25rem;
    color: #0077b6;
  }
