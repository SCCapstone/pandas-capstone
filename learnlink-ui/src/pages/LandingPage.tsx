// LandingPage.tsx
import React from 'react';
import Navbar from '../components/Navbar';
import './LandingPage.css';
import CopyrightFooter from '../components/CopyrightFooter';

const LandingPage: React.FC = () => {
  return (
    <div className="landing-page">
      <Navbar />
      <main className="content">
        <p>To start searching please update your profile via top left corner!</p>
      </main>
      <CopyrightFooter />
    </div>
  );
};

export default LandingPage;
