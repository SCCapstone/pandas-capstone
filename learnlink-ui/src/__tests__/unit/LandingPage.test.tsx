import { render, screen } from '@testing-library/react';
import LandingPage from '../../pages/LandingPage';
import '@testing-library/jest-dom'; // Import jest-dom for assertions

// Mock Navbar and CopyrightFooter to focus tests on the content of the page
jest.mock('../../components/Navbar', () => () => <div data-testid="navbar">Navbar</div>);
jest.mock('../../components/CopyrightFooter', () => () => <div data-testid="footer">Footer</div>);

describe('LandingPage', () => {
  test('renders Navbar component', () => {
    render(<LandingPage />);

    // Check if the Navbar is rendered
    const navbar = screen.getByTestId('navbar');
    expect(navbar).toBeInTheDocument();
  });

  test('renders CopyrightFooter component', () => {
    render(<LandingPage />);

    // Check if the CopyrightFooter is rendered
    const footer = screen.getByTestId('footer');
    expect(footer).toBeInTheDocument();
  });

  test('displays the correct informational message', () => {
    render(<LandingPage />);

    // Check if the informational message is displayed correctly
    const message = screen.getByText(/To start matching, please update your ideal match factor via the profile tab!/i);
    expect(message).toBeInTheDocument();
  });

  test('renders the main content section', () => {
    render(<LandingPage />);

    // Check if the main content section exists
    const mainContent = screen.getByRole('main');
    expect(mainContent).toBeInTheDocument();
  });

});
