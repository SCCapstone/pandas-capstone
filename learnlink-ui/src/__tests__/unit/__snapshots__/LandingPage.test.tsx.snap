// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="landing-page"
  >
    <div>
      <div
        data-testid="navbar"
      >
        Navbar
      </div>
    </div>
    <main
      class="content"
    >
      <p>
        To start matching, please update your ideal match factor via the profile tab!
      </p>
    </main>
    <div>
      <div
        data-testid="footer"
      >
        Footer
      </div>
    </div>
  </div>
</DocumentFragment>
`;
